endpoint_info.go

```diff
@@ -159,7 +159,11 @@ func (ep *endpoint) Info() EndpointInfo {
 		return ep
 	}
 
-	return sb.getEndpoint(ep.ID())
+	if epi := sb.getEndpoint(ep.ID()); epi != nil {
+		return epi
+	}
```

Comment:
In case `getEndpoint` returns a nil endpoint, `Info()` will return a _non-nil_ `EndpointInfo` backed by a nil endpoint.
The caller nil check against the interface will be negative and further method invocation on `EndpointInfo` will result in nil pointer deference.